vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -105,9 +105,36 @@ func NewVolumeDriver(targetURL string, projectID string, hostID string, mountDir
 
 	return d
 }
-func (d *VolumeDriver) getRefCount(vol string) uint           { return d.refCounts.GetCount(vol) }
-func (d *VolumeDriver) incrRefCount(vol string) uint          { return d.refCounts.Incr(vol) }
-func (d *VolumeDriver) decrRefCount(vol string) (uint, error) { return d.refCounts.Decr(vol) }
+
+// In following three operations on refcount, if refcount
+// map hasn't been initialized, return 1 to be safe.
+
+// Return the number of references for the given volume
+func (d *VolumeDriver) getRefCount(vol string) uint {
+	if d.refCounts.GetInitSuccess() != true {
+		return 1
+	}
+
+	return d.refCounts.GetCount(vol)
+}
+
+// Increment the reference count for the given volume
+func (d *VolumeDriver) incrRefCount(vol string) uint {
+	if d.refCounts.GetInitSuccess() != true {
+		return 1
+	}
+
+	return d.refCounts.Incr(vol)
+}
+
+// Decrement the reference count for the given volume
+func (d *VolumeDriver) decrRefCount(vol string) (uint, error) {
+	if d.refCounts.GetInitSuccess() != true {
+		return 1, nil
```

Comment:
Its not really an error in this case. We just want to hold the refcount so that volume doesn't get deleted till refcounting succeeds. There could be error later on, hence the return type.